python-avd/pyavd/_eos_designs/structured_config/base/__init__.py

```diff
@@ -163,8 +163,35 @@ def router_multicast(self) -> None:
 
     @structured_config_contributor
     def hardware_counters(self) -> None:
-        """hardware_counters set based on hardware_counters.features variable."""
-        self.structured_config.hardware_counters = self.inputs.hardware_counters
+        """
+        Set hardware_counters.
+
+        Contributing data sources:
+          - hardware_counters.features variable.
+          - platform_settings.feature_support.hardware_counters fact.
+        """
+        if not self.inputs.hardware_counters:
+            return
+        # Avoid collision with eos_cli_config_gen
+        if not self.shared_utils.platform_settings.feature_support.hardware_counters.supported:
+            self.custom_structured_configs.nested.hardware_counters = EosCliConfigGen.HardwareCounters._from_null()
+            return
+        hardware_counters = self.inputs.hardware_counters._cast_as(EosCliConfigGen.HardwareCounters)
+
+        # Filter different hardware counter features based on the platform supportability
+        if self.shared_utils.platform_settings.feature_support.hardware_counters.feature.supported:
+            hardware_counters.features = hardware_counters.features._filtered(
+                lambda feature: get_v2(
+                    self.shared_utils.platform_settings.feature_support.hardware_counters.feature,
+                    feature.name.replace(" ", "_").replace("-", "_"),
+                    # Assume all uncovered/new features are supported
+                    default=True,
+                )
+            )
+        elif hasattr(hardware_counters, "features"):
```

Comment:
I don't think this check will work as you expect since it is a class. It will always delete it. (This is because we override getattr in our classes)
```suggestion
        elif hardware_counters.features:
```

Reply:
Fixed